integration/tests/neon_evm/test_neon_core_rest_api.py

```diff
@@ -1,58 +1,85 @@
+import base58
+import pytest
 from eth_utils import abi, to_text
 
 from .utils.contract import deploy_contract, get_contract_bin
 
 
+def check_account_info(account_info, response):
+    def decode_pubkey(pubkey):
+        return base58.b58encode(bytes(pubkey)).decode("utf-8")
+
+    assert "accounts_data" in response
+    assert account_info in [None, "Changed", "All"]
+
+    if account_info in ["Changed", "All"]:
+        assert len(response["accounts_data"]) > 0
+        assert len(response["solana_accounts"]) > 0
+
+        all_accounts = [account["pubkey"] for account in response["solana_accounts"]]
+        writable_accounts = [account["pubkey"] for account in response["solana_accounts"] if account["is_writable"]]
+
+        actual_accounts = [decode_pubkey(account["pubkey"]) for account in response["accounts_data"]]
+
+        if account_info == "Changed":
+            assert set(actual_accounts) == set(writable_accounts)
+            assert set(actual_accounts) != set(all_accounts)
+            assert set(actual_accounts).issubset(set(all_accounts))
+        elif account_info == "All":
+            assert set(actual_accounts) == set(all_accounts)
```

Comment:
actual_accounts and all_accounts created based on a response
It is not good
It is better to prepare expected list of accounts based on certain transaction

Reply:
Refactored ✅